Update docs for errors (#3618)
tarantool/doc: doc/reference/reference_lua/box_error/new.rst

box.error.new()
===============================================================================

.. _box_error-new-error-table:

.. function:: box.error.new({ reason = string[, code = number, type = string] })

>     Create the error defined by the specified parameters.

Reviewer: ```suggestion
    Create an error object with the specified parameters.
```

To avoid unnecessary passive voice.
Also, the point that this method does not raise the error looks useful to me here.
